internal/controller/rsa_controller.go

```diff
@@ -0,0 +1,85 @@
+package controller
+
+import (
+	"context"
+
+	"github.com/belastingdienst/opr-paas/internal/crypt"
+	"github.com/rs/zerolog/log"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var (
+	// crypts contains a maps of crypt against a Paas name
+	crypts             map[string]*crypt.Crypt
+	decryptPrivateKeys *crypt.CryptPrivateKeys
+)
+
+// resetCrypts removes all crypts and resets decryptSecretPrivateKeys
+func resetCrypts() {
+	crypts = nil
+	decryptPrivateKeys = nil
+}
+
+// getOrEnsureRsaSecret ensures that secret exists creating an empty secret if needed
+// and returns the body of the fetched or created secret
+func (r *PaasNSReconciler) getOrEnsureRsaSecret(
+	ctx context.Context,
+) (keys *crypt.CryptPrivateKeys, err error) {
+	// See if rsa secret exists and create if it doesn't
+	rsaSecret := &corev1.Secret{}
+	config := GetConfig()
+	namespacedName := config.DecryptKeysSecret
+
+	err = r.Get(ctx, types.NamespacedName{
+		Name:      namespacedName.Name,
+		Namespace: namespacedName.Namespace,
+	}, rsaSecret)
+	if err != nil && errors.IsNotFound(err) {
```

Comment:
```suggestion
	if err != nil {
```
Any error would be an issue to report back to the controller.